### DISCUSSION TITLE
Feature request: Move CSS properties into pseudoelement

### DISCUSSION THREAD

**Discussion by lhunath** [FortAwesome/Font-Awesome#20923] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Currently, the FontAwesome CSS applies font and text properties onto the element that the `fa-` classes are set to. This creates a `::before` (and sometimes `::after`) pseudo-element, where FontAwesome unicode content is placed. The current CSS relies on the pseudo-elements inheriting the actual properties they need from the parent element.

Unfortunately, this also overrides the text behaviour of the element itself, meaning any text present inside the element is changed to accomodate the FontAwesome font and text properties, making it impossible to do things like:
```
<a class="fa-solid fa-at">email@me.com</a>
```
And style the link text properly.

**Describe the solution you'd like**
I'd like to request moving all relevant FontAwesome properties into the pseudo-element so that the main element itself is no longer clobbered by them, leaving it free to be styled for other purposes. This also brings the properties closer to where they are actually relevant.

**Describe alternatives you've considered**
Using dedicated elements:
```
<a><i class="fa-solid fa-at"></i> email@me.com</a>
```
Unfortunately, this is more of a work-around than a solution, since it doesn't avoid element clobbering of properties that are really only used by pseudo-elements, and this solution doesn't work universally.  Sometimes we want the icon to be attached to the element for structural reasons, sometimes the element in question doesn't allow for internal HTML.

**Feature request checklist**
- [x] This is a single feature (i.e. not a re-write of all of Font Awesome)
- [x] The title starts with "Feature request: " and is followed by a clear feature name (Ex: `Feature request: moar cowbell`)
- [x] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome-Pro/issues) and to the best of my knowledge this is not a duplicate

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community.

This has been proposed and discussed before, and comes up with time to time. Last time I checked, this had side effects with IE. I think this will not be a problem anymore. Will try to find the original request when I have more time

Also, using pseudo-elements is discouraged by docs (ref: https://fontawesome.com/how-to-use/on-the-web/advanced/css-pseudo-elements)

I would like to keep the implementation simple and leave to developers the chance to move style to pseudo-elements, if the default Font Awesome approach does not fit their use case

@robmadole any thoughts?
